File: FurnitureStore/FurnitureStore.Models/WishList/WishList.cs

namespace FurnitureStore.Models.WishList
{
    public class WishList
    {
        public long Id { get; set; }
        public DateTime DateCreated { get; set; }
        public  string CustomerId { get; set; }
        public virtual User.User Customer { get; set; }
        public virtual ICollection<Product.Product> Products { get; set; } = new HashSet<Product.Product>();

    }
}

File: FurnitureStore/UI/furniturestore_mobile/lib/models/wish_list/wish_list.dart
import 'package:furniturestore_mobile/models/product/product.dart';
import 'package:json_annotation/json_annotation.dart';

part 'wish_list.g.dart';

@JsonSerializable()
class WishListModel {
  final int id;
  final DateTime dateCreated;
  final String customerId;
  final List<ProductModel> products;

  WishListModel(
      {required this.id,
      required this.dateCreated,
      required this.customerId,
      required this.products});

  factory WishListModel.fromJson(Map<String, dynamic> json) =>
      _$WishListModelFromJson(json);

  Map<String, dynamic> toJson() => _$WishListModelToJson(this);
}
